sdk/core/core-util/src/delay.ts

```diff
@@ -31,3 +32,29 @@ export function delay(timeInMs: number, options?: DelayOptions): Promise<void> {
     },
   );
 }
+
+/**
+ * Calculates the delay interval for retry attempts using exponential delay with jitter.
+ * @param retryAttempt - The current retry attempt number.
+ * @param config - The exponential retry configuration.
+ * @returns An object containing the calculated retry delay.
+ */
+export function calculateExponentialDelayInterval(
+  retryAttempt: number,
+  config: {
+    retryDelayInMs: number;
+    maxRetryDelayInMs: number;
+  },
+): { retryAfterInMs: number } {
```

Comment:
any benefit of returning an object?

Reply:
Just backwards compatibility in case we want to add more properties in the future in a non-breaking way (if we just returned the number, adding more properties would be a breaking change)